### PR TITLE
Movement Optimisation

### PR DESCRIPTION
**Pull request by NanakoAC**
A major step in reducing movement lag:

This PR gives each turf an internal list of things it contains which could possibly block movement, based on a variable set on atoms.

Only things in that list are checked for collisions.

This saves at least one check on every tile by not trying to collide with lighting overlays. It saves many more in cases of pipes, wires, catwalks, corruption, decals etc. Things which are in every room.
And it saves vast quantities of checks in cluttered areas littered with trash, bodyparts, blood, et al.

Basically, floors that don't contain dense stuff no longer generate collision events at all

We had 48 million calls to canpass in the most recent data, this will help

Also supplemental change: Explosions no longer break power cables. This was an unfun feature to have

### CODE REVIEW

**Comment by AshTheGuy** [comment]
Based?

**Comment by AshTheGuy** [comment]
> Based?

Dont think its based at all, shit now lags 100x worse. think you messed up a line while coding.
